src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1437,7 +1453,7 @@ def _process_decln(self, parent, symbol_table, decl, default_visibility,
             base_type = ScalarType(data_name, precision)
         elif isinstance(type_spec, Fortran2003.Declaration_Type_Spec):
             # This is a variable of derived type
-            type_name = str(walk(type_spec, Fortran2003.Type_Name)[0])
+            type_name = str(walk(type_spec, Fortran2003.Type_Name)[0]).lower()
```

Comment:
It was necessary but actually, as discussed in Teams, this was because of a bug in `SymbolTable.remove()`. I've fixed the bug, added tests and removed this `lower()`.